### PR TITLE
#638/mapk docs

### PR DESCRIPTION
**Pull request by rcd27**
Changed indents in snippets, added `main()` function to `ank:playground` snippet

### CODE REVIEW

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 1]
```diff
@@ -30,64 +30,69 @@ import arrow.core.MapK
...
-    val arrowMap: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+  val arrowMap: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
 ```
 
 You can modify each map entry value and get another MapK with use of `map(..)` function:
 
 ```kotlin:ank
-    val arrowMap: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
-    val modifiedMap: MapK<String, Int> = arrowMap.map { entryValue -> entryValue.plus(10) }
+
```
Comment (pakoito) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]: Extra whitespace

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 2]
```diff
@@ -30,64 +30,64 @@ import arrow.core.MapK
...
-    println(foldLeft)
-    val foldRight: Eval<String> =
-        map1.foldRight(Eval.just("one")) { entry, eval ->
-            Eval.just("$entry ${eval.value()}")
-        }
-    println(foldRight)
+val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+
+val foldLeft: String = map1.foldLeft("one") { entry, tuple -> entry + tuple }
+println(foldLeft)
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]: put this `println(foldLeft)` at the end of the snippet

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 3]
```diff
@@ -30,64 +30,64 @@ import arrow.core.MapK
...
-    val flattened: MapK<String, String> = map1.flatMap { map2 }
+val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+val map2: MapK<String, String> = mapOf("one" to "ONE", "three" to "three").k()
+
+val flattened: MapK<String, String> = map1.flatMap { map2 }
 ```
 
 `foldLeft` and `foldRight` are used for element aggregation:
 
 ```kotlin:ank
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
```kotlin:ank:playground
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 4]
```diff
@@ -30,64 +30,64 @@ import arrow.core.MapK
...
-    val map2: MapK<String, (Int) -> String> = mapOf("one" to f1, "two" to f2).k()
-    
-    val apResult = map1.ap(map2)
+val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+
+val f1: (Int) -> String = { i: Int -> "f1 to \"$i\"" }
+val f2: (Int) -> String = { i: Int -> "f2 to \"$i\"" }
+val map2: MapK<String, (Int) -> String> = mapOf("one" to f1, "two" to f2).k()
+
+val apResult = map1.ap(map2)
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
val apResult = map1.ap(map2)
apResult
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 5]
```diff
@@ -30,64 +30,64 @@ import arrow.core.MapK
...
-    val secondBag: MapK<String, Int> = mapOf("eggs" to 6, "cheese" to 1).k()
-    
-    val eggsBag: MapK<String, Int> = firstBag.map2(secondBag) { firstBagMatch, secondBagMatch ->
-        // If there are matching keys
-        firstBagMatch + secondBagMatch     // you can modify the value of output MapK
-    }
+val firstBag: MapK<String, Int> = mapOf("eggs" to 5, "milk" to 1).k()
+val secondBag: MapK<String, Int> = mapOf("eggs" to 6, "cheese" to 1).k()
+
+val eggsBag: MapK<String, Int> = firstBag.map2(secondBag) { firstBagMatch, secondBagMatch ->
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```
val eggsBag: MapK<String, Int> = firstBag.map2(secondBag) { firstBagMatch, secondBagMatch ->
  // If there are matching keys
  firstBagMatch + secondBagMatch     // you can modify the value of output MapK
}
```
check indent's

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 6]
```diff
@@ -30,64 +30,64 @@ import arrow.core.MapK
...
+val arrowMap: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
 ```
 
 You can modify each map entry value and get another MapK with use of `map(..)` function:
 
 ```kotlin:ank
-    val arrowMap: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
-    val modifiedMap: MapK<String, Int> = arrowMap.map { entryValue -> entryValue.plus(10) }
+val arrowMap: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+val modifiedMap: MapK<String, Int> = arrowMap.map { entryValue -> entryValue.plus(10) }
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
val modifiedMap: MapK<String, Int> = arrowMap.map { entryValue -> entryValue.plus(10) }
modifiedMap
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 7]
```diff
@@ -101,14 +101,16 @@ import arrow.core.fix
...
 
-    val optionMap = mapOf(1.some() to "one", 2.some() to "two", None to "none").k()
-            .traverse(Option.applicative()) { value ->
-                when (value) {
-                    "one", "two", "none" -> Some(value)
-                    else -> None
-                }
-            }.fix()
-        println(optionMap)
+fun main(args: Array<String>) {
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
fun main() {
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 8]
```diff
@@ -101,14 +104,16 @@ import arrow.core.fix
...
-    val optionMap = mapOf(1.some() to "one", 2.some() to "two", None to "none").k()
-            .traverse(Option.applicative()) { value ->
-                when (value) {
-                    "one", "two", "none" -> Some(value)
-                    else -> None
-                }
-            }.fix()
-        println(optionMap)
+fun main() {
+val optionMap = mapOf(1.some() to "one", 2.some() to "two", None to "none").k()
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]: check Indentation here too

File: modules/docs/arrow-docs/build.gradle [hunk 9]
```diff
@@ -15,7 +15,7 @@ dependencies {
     compile project(':arrow-fx-rx2')
     compile project(':arrow-fx-reactor')
     compile project(':arrow-optics-mtl')
-    compile project(':arrow-validation')
+//    compile project(':arrow-validation')
```
Comment (i-walker) [inline modules/docs/arrow-docs/build.gradle]: ? why 

change that back

File: settings.gradle [hunk 10]
```diff
@@ -108,9 +108,9 @@ include {
             _ 'reflect'
         }
 
-        validation {
-            _ 'validation'
-        }
+//        validation {
+//            _ 'validation'
```
Comment (i-walker) [inline settings.gradle]: same here

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 11]
```diff
@@ -30,64 +30,74 @@ import arrow.core.MapK
...
 
 ```kotlin:ank
-    val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
-    val map2: MapK<String, String> = mapOf("one" to "ONE", "three" to "three").k()
-    
-    val flattened: MapK<String, String> = map1.flatMap { map2 }
+val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+val map2: MapK<String, String> = mapOf("one" to "ONE", "three" to "three").k()
+
+val flattened: MapK<String, String> = map1.flatMap { map2 }
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
val flattened: MapK<String, String> = map1.flatMap { map2 }
flattended
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md (hunk 5 above)
Comment (i-walker) [reply modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]: not resolved IMO

File: settings.gradle (hunk 10 above)
Comment (rcd27) [reply settings.gradle]: yea, forgot to revert. fixed

File: modules/docs/arrow-docs/build.gradle (hunk 9 above)
Comment (rcd27) [reply modules/docs/arrow-docs/build.gradle]: build was failing, changed back

**Comment by rcd27** [comment]
Seems like one test doesn't pass

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 12]
```diff
@@ -30,64 +30,74 @@ import arrow.core.MapK
...
-        }
-    println(foldRight)
+```kotlin:ank:playground
+import arrow.core.Eval
+import arrow.core.MapK
+import arrow.core.k
+
+
+fun main() {
+  val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
  //sampleStart
  val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 13]
```diff
@@ -30,64 +30,74 @@ import arrow.core.MapK
...
+
+  val foldLeft: String = map1.foldLeft("one") { entry, tuple -> entry + tuple }
+  val foldRight: Eval<String> =
+    map1.foldRight(Eval.just("one")) { entry, eval ->
+      Eval.just("$entry ${eval.value()}")
+    }
+
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
  //sampleEnd
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 14]
```diff
@@ -101,14 +111,16 @@ import arrow.core.fix
...
-        println(optionMap)
+fun main() {
+  val optionMap = mapOf(1.some() to "one", 2.some() to "two", None to "none").k()
+    .traverse(Option.applicative()) { value ->
+      when (value) {
+        "one", "two", "none" -> Some(value)
+        else -> None
+      }
+    }.fix()
+  println(optionMap)
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
  //sampleEnd
  println(optionMap)
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 15]
```diff
@@ -101,14 +111,16 @@ import arrow.core.fix
...
-    val optionMap = mapOf(1.some() to "one", 2.some() to "two", None to "none").k()
-            .traverse(Option.applicative()) { value ->
-                when (value) {
-                    "one", "two", "none" -> Some(value)
-                    else -> None
-                }
-            }.fix()
-        println(optionMap)
+fun main() {
+  val optionMap = mapOf(1.some() to "one", 2.some() to "two", None to "none").k()
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
  //sampleStart
  val optionMap = mapOf(1.some() to "one", 2.some() to "two", None to "none").k()
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 16]
```diff
@@ -30,64 +30,76 @@ import arrow.core.MapK
...
-        // If there are matching keys
-        firstBagMatch + secondBagMatch     // you can modify the value of output MapK
-    }
+val firstBag: MapK<String, Int> = mapOf("eggs" to 5, "milk" to 1).k()
+val secondBag: MapK<String, Int> = mapOf("eggs" to 6, "cheese" to 1).k()
+
+val eggsBag: MapK<String, Int> = firstBag.map2(secondBag) { firstBagMatch, secondBagMatch ->
+  // If there are matching keys
+  firstBagMatch + secondBagMatch  // you can modify the value of output MapK
+  }
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
}
```

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 17]
```diff
@@ -30,64 +30,76 @@ import arrow.core.MapK
...
 ```kotlin:ank
-    val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
-    val map2: MapK<String, String> = mapOf("one" to "ONE", "three" to "three").k()
-    
-    val flattened: MapK<String, String> = map1.flatMap { map2 }
+val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+val map2: MapK<String, String> = mapOf("one" to "ONE", "three" to "three").k()
+
+val flattened: MapK<String, String> = map1.flatMap { map2 }
+flattended
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
flattened
```
Typo,  sorry :)

File: modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md [hunk 18]
```diff
@@ -30,64 +30,76 @@ import arrow.core.MapK
...
-    val foldRight: Eval<String> =
-        map1.foldRight(Eval.just("one")) { entry, eval ->
-            Eval.just("$entry ${eval.value()}")
-        }
-    println(foldRight)
+```kotlin:ank:playground
+import arrow.core.Eval
+import arrow.core.MapK
+import arrow.core.k
+
```
Comment (i-walker) [inline modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md]:
```suggestion
```
delete extra line

**Pull request merged**